WordPress/src/main/java/org/wordpress/android/viewmodel/giphy/provider/TenorProvider.kt

```diff
@@ -0,0 +1,185 @@
+package org.wordpress.android.viewmodel.giphy.provider
+
+import android.content.Context
+import android.net.Uri
+import com.tenor.android.core.constant.AspectRatioRange
+import com.tenor.android.core.constant.MediaCollectionFormat
+import com.tenor.android.core.constant.MediaFilter
+import com.tenor.android.core.model.impl.Result
+import com.tenor.android.core.network.ApiClient
+import com.tenor.android.core.network.ApiService
+import com.tenor.android.core.network.IApiClient
+import com.tenor.android.core.response.WeakRefCallback
+import com.tenor.android.core.response.impl.GifsResponse
+import org.wordpress.android.BuildConfig
+import org.wordpress.android.R.string
+import org.wordpress.android.viewmodel.giphy.GiphyMediaViewModel
+import org.wordpress.android.viewmodel.giphy.MutableGiphyMediaViewModel
+import org.wordpress.android.viewmodel.giphy.provider.GifProvider.GifRequestFailedException
+
+/**
+ * Implementation of a GifProvider using the Tenor GIF API as provider
+ *
+ * This Provider performs requests to the Tenor API using the [tenorClient].
+ */
+
+internal class TenorProvider @JvmOverloads constructor(
+    val context: Context,
+    tenorClient: IApiClient? = null
```

Comment:
This field was created to make an easy opening for unit tests but allowing it as an optional value without disregard the API initialization. I think the change at 0b7035d8729623f30414a9f91a156b0c414732f2 already solved this point since the initialization of the API is done outside the TenorProvider, at the ApplicationModule, now the injection of the IApiClient happens there right after the API key configuration.

Reply:
good point about the unit tests, I like the way it's done now